### PR TITLE
Feat/health

### PR DESCRIPTION
**Pull request by prawl**
Can now toggle all survival stats

### CODE REVIEW

File: app/models/character.rb [hunk 1]
```diff
@@ -19,6 +19,16 @@ def tick
...
+  def restore_health(restore)
+    self.health += restore
+    if self.health > 100
+      return 100
+    else
+      self.health
+    end
+    save
+  end
+
```
Comment (dkniffin) [inline app/models/character.rb]: For all these methods, let's rewrite them like this:

````ruby
  def restore_health(restore)
    new_health = health + restore
    new_health = 100 if new_health > 100
    update(health: new_health)
  end
````

**Comment by dkniffin** [comment]
Other than the one comment above, the only other change I'd like before merging is to add tests for each of the model methods. Otherwise, :+1:

File: app/models/character.rb (hunk 1 above)
Comment (prawl) [inline app/models/character.rb]: :+1:

**Comment by prawl** [comment]
Updated!

File: spec/models/character_spec.rb [hunk 2]
```diff
@@ -18,21 +18,79 @@
   end
 
   describe '#take_damage' do
```
Comment (dkniffin) [inline spec/models/character_spec.rb]: Let's add a `before { subject.update(health: 100) }` here. I know the factory sets it up like that, but I don't like depending on the data from the factory.

**Comment by dkniffin** [comment]
One more change, then :shipit:

File: spec/models/character_spec.rb (hunk 2 above)
Comment (prawl) [inline spec/models/character_spec.rb]: :+1:

**Pull request merged**